pkg/virtctl/console/console.go

```diff
@@ -209,82 +107,63 @@ func TerminalWebsocketCallback(ws *websocket.Conn, resp *http.Response, connErr
 		close(stopChan)
 	}()
 
-	// If there is no obvious connection error, set up the terminal
-	if connErr == nil {
-		state, err := terminal.MakeRaw(int(os.Stdin.Fd()))
-		if err != nil {
-			return fmt.Errorf("Make raw terminal failed: %s", err)
+	go func() {
+		defer close(readStop)
+		buf := make([]byte, 1024, 1024)
+		for {
+			// reading stdout from console
+			n, err := stdoutReader.Read(buf)
+			if err != nil && err != io.EOF {
+				return
+			}
+			if n == 0 && err == io.EOF {
+				return
+			}
+			// Writing to stdout
+			_, err = out.Write(buf[0:n])
+			if err == io.EOF {
+				return
+			}
 		}
-		defer terminal.Restore(int(os.Stdin.Fd()), state)
-		fmt.Fprint(os.Stderr, "Escape sequence is ^]")
-	}
-
-	return NewWebsocketCallback(os.Stdin, os.Stdout, stopChan)(ws, resp, connErr)
-}
-
-func RequestFromConfig(config *rest.Config, vm string, namespace string, device string) (*http.Request, error) {
+	}()
 
-	u, err := url.Parse(config.Host)
-	if err != nil {
-		return nil, err
-	}
+	go func() {
+		defer close(writeStop)
+		buf := make([]byte, 1024, 1024)
+		for {
+			// reading from stdin
```

Comment:
`io.Copy` can be used. See https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-api/rest/console.go#L123.

Reply:
fixed

Reply:
actually I had to revert this. This is the one case where i'm not using io.copy now because we have to read the escape sequence.